kura/org.eclipse.kura.http.server.manager/src/main/java/org/eclipse/kura/http/server/manager/HttpService.java

```diff
@@ -97,22 +101,33 @@ private synchronized void restartHttpService() {
     }
 
     private synchronized void activateHttpService() {
-        try {
-            logger.info("starting Jetty instance...");
-            this.jettyServerHolder = new JettyServerHolder(this.options, Optional.ofNullable(this.keystoreService),
-                    this.dispatcherServlet, this.eventListener);
-            logger.info("starting Jetty instance...done");
-        } catch (final Exception e) {
-            logger.error("Could not start Jetty Web server", e);
-        }
+        this.executorService.submit(() -> {
+            try {
+                logger.info("starting Jetty instance...");
+                this.jettyServerHolder = new JettyServerHolder(this.options, Optional.ofNullable(this.keystoreService),
+                        this.dispatcherServlet, this.eventListener);
+                logger.info("starting Jetty instance...done");
+            } catch (final Exception e) {
+                logger.error("Could not start Jetty Web server", e);
+            }
+        });
     }
 
     private synchronized void deactivateHttpService() {
         try {
             logger.info("stopping Jetty instance...");
-            this.jettyServerHolder.stop();
+            if (this.jettyServerHolder != null) {
+                this.jettyServerHolder.stop();
+            }
+            this.executorService.shutdown();
```

Comment:
We should probably shutdown the executor and wait its termination in the deactivate() method instead of here.